src/bigquery.ts

```diff
@@ -742,9 +799,13 @@ export class BigQuery extends common.Service {
    * @returns {string} The type detected from the value.
    */
   // tslint:disable-next-line no-any
-  static getType_(value: any): ValueType {
+  static getType_(value: any, providedType?: any): ValueType {
     let typeName;
 
+    if (providedType) {
+      return this.getProvidedType_(providedType);
```

Comment:
The naming here confused me a bit. Just for consideration, what about:

```diff
- getType_
+ getTypeDescriptorFromValue_

- getProvidedType_
+ getTypeDescriptorFromProvidedType_
```

Definitely longer, but I think something like that would help someone jump in for a bug fix more quickly.

Reply:
Done.